source/common/quic/envoy_quic_utils.cc

```diff
@@ -149,6 +151,10 @@ createConnectionSocket(const Network::Address::InstanceConstSharedPtr& peer_addr
   }
   connection_socket->addOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
   connection_socket->addOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.prefer_udp_gro") &&
```

Comment:
I'm a bit concerned about the runtime guard value to be inconsistent here and in the connection. How complicated would it be to change this function to take a bool which latches the value of the runtime guard in the connection?

Reply:
Unfortunately, there isn't a very clean way to do this, because `createConnectionSocket` gets called in the `EnvoyQuicClientConnection` constructor, and that constructor actually calls into another `EnvoyQuicClientConnection` constructor in the initialization list, so we can't initialize a variable (e.g. prefer_gro) in the same initialization list (you get compiler errors).  We can inline the call to `Runtime::runtimeFeatureEnabled("envoy.reloadable_features.prefer_udp_gro")` when calling `createConnectionSocket`, but it will still result in two calls to `Runtime::runtimeFeatureEnabled`, so I'm not sure what the net benefit is.  It would actually be 3 calls b/c we'd have to set the runtime guard prefer_gro value in connection migration too (https://github.com/envoyproxy/envoy/blob/e28e0d67fc9f9f677b644f940320c5c01bf057c9/source/common/quic/envoy_quic_client_connection.cc#L157).

So I'm inclined to think it's not worth it, unless you had some other concern in mind?

Reply:
Dan is your concern values being out of sync or the cost of checking flags?  Either way we could make it a restart flag (to avoid syncing issues) and latch if we have perf concerns.

Reply:
I was originally concerned about flag values out of sync, but cost of checking the flag once per event loop might also be an issue. How about changing to a restart flag and latching the value in the connection?

Reply:
flag is now passed in from client_connection_factory_impl